File: java/PinkFluffyUnicornDancingOnRainbow.java
public class PinkFluffyUnicornDancingOnRainbow {

    private final long RATE = 100;
    private final String[] OPEN_UNICORN = new String[]{
          "          /| ",
          " ~～～～~/ | ",
          "|       O  O ",
          "|         ∇ |",
          " ~~~~∪∪~~~∪∪ "
        };
    private final String[] CLOSE_UNICORN = new String[]{
          "          /| ",
          " ～～～～/ | ",
          "|       ⌒  ⌒ ",
          "|         ∇ |",
          " ~~∪∪~~~∪∪~~ "
        };
    private int[] COLORS = new int[]{31, 33, 32, 36, 35, 34};
    private final int UNICORN_WIDTH = OPEN_UNICORN[0].length();

    private int termWidth = 64;
    private int unicornOffset = (termWidth - UNICORN_WIDTH) / 2;

    private long counter = 0;

    public static void main(String[] args) throws Exception {
        PinkFluffyUnicornDancingOnRainbow unicorn = new PinkFluffyUnicornDancingOnRainbow();
        unicorn.run();
    }

    public void run() throws Exception {
        while (true) {
            clear();

            draw(counter);
            counter++;

            println("Press Ctrl-C to exit...");

            Thread.sleep(RATE);
        }
    }

    private void print(Object _) {
        if (_ == null) return;
        System.out.print(_);
    }

    private void println(Object _) {
        if (_ == null) return;
        System.out.println(_);
    }

    private void clear() {
        print("\033c");
    }

    private void renderSpace(int number) {
        for (int i = 0; i < number; i++) {
            print(" ");
        }
    }

    private String pinkify(String _) {
        return "\033[96m" + _ + "\033[0m";
    }

    private void renderUnicorn(String[] unicorns) {
        for (int i = 0; i < unicorns.length; i++) {
            renderSpace(unicornOffset);

            println(pinkify(unicorns[i]));
        }
    }

    private String rainbowify(String str, int index) {
        int color = COLORS[index];
        return "\033[" + color + "m" + str + "\033[0m";
    }

    private void renderRainbow() {
        for (int i = 0; i < 5; i++) {
            for (int j = 0; j < termWidth; j++) {
                long _ = counter + j;
                if (_ % 2 == 0) {
                    print(rainbowify("-", i));
                } else {
                    print(rainbowify("_", i));
                }
            }
            println("");
        }
    }

    private void draw(long counter) {
        String[] _unicorn = (counter % 10 > 5)? OPEN_UNICORN : CLOSE_UNICORN;
        renderUnicorn(_unicorn);

        renderRainbow();
    }
}

File: pascal/PinkFluffyUnicornDancingOnRainbow.pas
Program PinkFluffyUnicornDancingOnRainbow;
uses Crt, sysutils;
const
  UNICORN_HEIGHT = 5;
  UNICORN_WIDTH = 21;
  SPACE_OFFSET = 13;
type
  unicorn_array = array [1..UNICORN_HEIGHT] of string[UNICORN_WIDTH];
const
  RATE = 100;
  COLORS_COUNT = 5;
  OPEN_UNICORN : unicorn_array = (
          '          /| ',
          ' ~～～～~/ | ',
          '|       O  O ',
          '|         ∇ |',
          ' ~~~~∪∪~~~∪∪ '
  );
  CLOSE_UNICORN : unicorn_array = (
          '          /| ',
          ' ～～～～/ | ',
          '|       ⌒  ⌒ ',
          '|         ∇ |',
          ' ~~∪∪~~~∪∪~~ '
  );
  COLORS : array [1..COLORS_COUNT] of byte = (LightRed, Yellow, LightGreen, LightCyan, Magenta);
var
  term_width : integer = 64;
  counter : integer = 0;
  unicorn_offset : integer;

procedure RenderSpace (_offset : integer);
var i : integer = 0;
begin
  for i := 0 to _offset do
    Write(' ');
end;

procedure Pinkify (_ : string);
begin
  Textcolor(LightMagenta);
  WriteLn(_);
end;

procedure RenderUnicorn (_unicorns : unicorn_array);
var i : integer = 0;
begin
  for i := 0 to UNICORN_HEIGHT do
  begin
    RenderSpace(unicorn_offset);
    Pinkify(_unicorns[i]);
  end;
end;

procedure Rainbowify (_: string; index : integer);
begin
  Textcolor(COLORS[index]);
  Write(_);
end;

procedure RenderRainbow (_counter : integer);
var i, j : integer;
begin
  for i := 0 to (COLORS_COUNT) do
  begin
    for j := 0 to (term_width - 1) do
    begin
      if (_counter + j) mod 2 = 0 then
        Rainbowify('-', i)
      else
        Rainbowify('_', i);
    end;
    WriteLn('');
  end;
end;

procedure Draw (_counter : integer);
begin
  if _counter mod 10 > 5 then
    RenderUnicorn(OPEN_UNICORN)
  else
    RenderUnicorn(CLOSE_UNICORN);

  RenderRainbow(_counter);
end;

Begin
  unicorn_offset := round((term_width - UNICORN_WIDTH) / 2);

  while (true) do
  begin
    clrscr;

    Draw(counter);
    counter += 1;

    Textcolor(White);
    WriteLn('Press Ctrl-C to exit...');

    Delay(RATE);

    if KeyPressed then
      if ReadKey = ^C then
        break;
  end;
End.
